### ISSUE TITLE
[FEA] Support index keywords in merge

### ISSUE THREAD

**Issue by mrocklin** [rapidsai/cudf#979]
It would be nice to implement the `left_index=` and `right_index=` keywords for `cudf.DataFrame.merge`.  This would be in support of indexed joins in dask-cudf, see https://github.com/rapidsai/dask-cudf/issues/76 by @randerzander 

```python
import pandas as pd
left = pd.DataFrame({'x': [1, 2, 3]}, index=[1, 2, 3])
right = pd.DataFrame({'y': [10, 20]}, index=[1, 2])
left.merge(right, left_index=True, right_index=True)
```

```
   x   y
1  1  10
2  2  20
```

```python
left.merge(right, left_on='x', right_index=True)
```

```
   x   y
1  1  10
2  2  20
```

I suspect that near-term one solution here would be to elevate the index to a series and then call a normal merge using the `left_on=` and `right_on=` keywords.

**Comment by mrocklin** [comment]
cc @thomcom @kkraus14

**Comment by kkraus14** [comment]
Resolved by #1052
